.github/workflows/compliance.yml

```diff
@@ -37,6 +37,8 @@ jobs:
         git config --global user.email "you@example.com"
         git config --global user.name "Your Name"
         git remote -v
+        # Ensure there's no merge commits in the PR
+        git rev-list --min-parents=2 --count origin/${BASE_REF}.. | grep 0
```

Comment:
just one comment: is error message friendly enough? This is a typical newbie mistake, help matters here.

Reply:
Yes, I was thinking about that. Any suggestions?

Reply:
something like `if ...error...; echo "Merge commits not allowed, do X and Y"; exit 1; fi`

Reply:
I was really happy with the one-liner :) let me see if I can keep it small

Reply:
> ```
> [ "$(git rev-list --min-parents=2 --count origin/${BASE_REF}..)" == "0" ] || echo "::error ..."
> ```

Will this exit with a status code `!=0` though? The Action won't fail otherwise.

Reply:
You can even shorten it, apparently `--merges` is equivalent to `--min-parents=2`.

Reply:
> something like `if ...error...; echo "Merge commits not allowed, do X and Y"; exit 1; fi`

Maybe "Merge commits not allowed, use rebase instead."?

Reply:
ah nice! will do!